tests/test_project_api.py

```diff
@@ -205,10 +205,13 @@ async def test_delete_project(
         )
 
         # Check that a project-related job exists
-        stm = select(ApplyWorkflow).join(Project).where(Project.id == p["id"])
-        res = await db.execute(stm)
-        jobs = list(res)
-        assert len(jobs) == 1
+        stm = select(ApplyWorkflow).where(ApplyWorkflow.project_id == p["id"])
```

Comment:
We should also test `project.job_list` here, and verify that it points to the same (number of) jobs.